mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -744,7 +744,13 @@ private[classification] class MultiClassSummarizer extends Serializable {
   def countInvalid: Long = totalInvalidCnt
 
   /** @return The number of distinct labels in the input dataset. */
-  def numClasses: Int = distinctMap.keySet.max + 1
+  def numClasses: Int = try {
+    distinctMap.keySet.max + 1
+  } catch {
+    case e: UnsupportedOperationException =>
```

Comment:
Are you trying to handle the case where the map is empty? why not handle it directly then?
Just needs one new line of code, like: `require(distinctMap.nonEmpty)`